dropwizard-jdbi/src/main/java/io/dropwizard/jdbi/DBIFactory.java

```diff
@@ -83,33 +88,104 @@ private SanerNamingStrategy() {
+     * with the configured health check; this method should not be overridden
+     * (instead, override {@link #configure(DBI, String)} and/or
+     * {@link #newInstance(ManagedDataSource)})
+     * 
+     * @param environment
+     * @param configuration
+     * @param dataSource
+     * @param name
+     * @return A fully configured {@link DBI} object
+     */
     public DBI build(Environment environment,
                      PooledDataSourceFactory configuration,
                      ManagedDataSource dataSource,
                      String name) {
-        final String validationQuery = configuration.getValidationQuery();
-        final DBI dbi = new DBI(dataSource);
+
+        // Create the instance
+        final DBI dbi = this.newInstance(dataSource);
+        
+        // Manage the data source that created this instance.
         environment.lifecycle().manage(dataSource);
+        
+        // Setup the required health checks.
+        final String validationQuery = configuration.getValidationQuery();
         environment.healthChecks().register(name, new DBIHealthCheck(
                 environment.getHealthCheckExecutorService(),
                 configuration.getValidationQueryTimeout().orElseGet(() -> Duration.seconds(5)),
                 dbi,
                 validationQuery));
+
+        // Setup logging.
         dbi.setSQLLog(new LogbackLog(LOGGER, Level.TRACE));
+        
+        // Setup the timing collector
         dbi.setTimingCollector(new InstrumentedTimingCollector(environment.metrics(),
                 new SanerNamingStrategy()));
+        
         if (configuration.isAutoCommentsEnabled()) {
             dbi.setStatementRewriter(new NamePrependingStatementRewriter(new ColonPrefixNamedParamStatementRewriter()));
         }
-        dbi.registerArgumentFactory(new GuavaOptionalArgumentFactory(configuration.getDriverClass()));
-        dbi.registerArgumentFactory(new OptionalArgumentFactory(configuration.getDriverClass()));
+
+        // Add the default argument and column mapper factories.
+        this.configure(dbi, configuration);
+
+        return dbi;
+    }
+    
+    /**
+     * This creates a vanilla DBI instance based on the specified data source;
+     * this can be overridden if required
+     * 
+     * @param dataSource
+     * @return
+     */
+    protected DBI newInstance(final ManagedDataSource dataSource) {
+        return new DBI(dataSource);
+    }
+
+    /**
+     * This method sets the default column mapper and argument factories
+     * <p>
+     * if this is overridden it is strongly recommend that
+     * {@code super.configure(dbi, configuration)} is invoked before any other
+     * changes are made if you intend to use the default as a base so that the
+     * customized argument settings will supersede the defaults
+     * </p>
+     * 
+     * @param dbi
+     * @param configuration
+     */
+    protected void configure(final DBI dbi, final PooledDataSourceFactory configuration) {
```

Comment:
I originally had configure taking a second parameter as a `String` (called driverClazz), however the configuration class also happens to be the factory. Should anyone extend `PooledDataSourceFactory` to have extra options in their YAML config, they may wish to have that extra data available to their own `configure` method. That was my thinking anyway. I can very easily swap this back to using just a plain `String`.

As for `databaseTimeZone`, that is taken care of by a method so configure doesn't access it directly.

In retrospect, it might be ideal to take that as a parameter to configure instead of the method override approach; however I don't want to break any existing code in the field that might be using an override of the database time zone method.